tests/components/vevor_heater/test_sensor.py

```diff
@@ -0,0 +1,212 @@
+"""Unit tests for Vevor BLE Heater sensors."""
+
+from unittest.mock import patch
+
+from bleak import BLEDevice
+from vevor_heater_ble.heater import (
+    HeaterError,
+    OperationalMode,
+    OperationalStatus,
+    PowerStatus,
+    VevorDevice,
+    VevorHeaterStatus,
+)
+
+from homeassistant.components.vevor_heater import DOMAIN
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+
+
+async def test_sensors_target_level(
+    enable_bluetooth: None, hass: HomeAssistant
+) -> None:
+    """Tests sensors when the heater is set to fixed level heating."""
+    status_to_set = VevorHeaterStatus(power_status=PowerStatus.OFF)
+
+    with patch(
+        "homeassistant.components.bluetooth.async_ble_device_from_address",
+        return_value=BLEDevice(
+            address="AB:CD:EF:01:23:45", name="FooBar", details=None, rssi=-1
+        ),
+    ):
+
+        def update_self_status(self, bledevice):
+            self.status = status_to_set
+
+        with patch.object(
+            VevorDevice, "refresh_status", autospec=True
+        ) as refresh_status:
+            refresh_status.side_effect = update_self_status
+
+            entry = MockConfigEntry(
+                domain=DOMAIN, unique_id="Foo", data={"address": "AB:CD:EF:01:23:45"}
+            )
+            entry.add_to_hass(hass)
+
+            assert await hass.config_entries.async_setup(entry.entry_id)
+
+            await hass.async_block_till_done()
+            status_to_set = VevorHeaterStatus(
+                power_status=PowerStatus.RUNNING,
+                operational_mode=OperationalMode.POWER_LEVEL,
+                operational_status=OperationalStatus.HEATING,
+                elevation=123.45,
+                target_power_level=7,
+                current_power_level=7,
+                input_voltage=12.34,
+                combustion_temperature=222,
+                room_temperature=25,
+                error=HeaterError.NO_ERROR,
+            )
+
+            await hass.data[DOMAIN][entry.entry_id].async_refresh()
```

Comment:
Please don't access hass.data in new tests. Instead use the async_fire_timed_changed test helper

Reply:
Done.

Reply:
It looks like this change was missed

Reply:
Removed.